test/blocks/chart/chart.test.js

```diff
@@ -464,4 +476,35 @@ describe('chart', () => {
 
     expect(lineSeriesOptions(series, firstDataset, units)).to.eql(expected);
   });
+
+  it('init returns undefined if missing type', async () => {
+    document.body.innerHTML = '<div class="chart"></div>';
+    const el = document.querySelector('.chart');
+    expect(init(el)).to.be.undefined;
+  });
+
+  it('init generates list chart', async () => {
+    const linkRel = '/drafts/data-viz/list.json';
+    document.body.innerHTML = `<div class="chart list"><div>Title</div><div>Subtitle</div><div><div><a href="https://data-viz--milo--adobecom.hlx.page${linkRel}"></a></div></div><div>Footnote</div></div>`;
+    const data = await readFile({ path: './mocks/listChartSingleTable.json' });
+    const parsedData = JSON.parse(data);
+    fetch.withArgs(linkRel).returns({ ok: true, json: () => parsedData });
+    const el = document.querySelector('.chart');
+    init(el);
+    const listWrapper = await waitForElement('.list-wrapper');
+    expect(listWrapper).to.exist;
+  });
+
+  it('init chart with echarts without intersection observer', async () => {
+    window.IntersectionObserver = undefined;
+    const linkRel = '/drafts/data-viz/bar.json';
+    document.body.innerHTML = `<div class="chart bar"><div>Title</div><div>Subtitle</div><div><div><a href="https://data-viz--milo--adobecom.hlx.page${linkRel}"></a></div></div><div>Footnote</div></div>`;
+    const data = await readFile({ path: './mocks/barChart.json' });
+    const parsedData = JSON.parse(data);
```

Comment:
Is `JSON.parse(data)` needed? `fetchData` uses `resp.json()` so I would think it's being parsed after the fetch but I'm not familiar with stubs.